### PR TITLE
#15: Fix support for .sk domains, tested on priamaakcia.sk

### PR DESCRIPTION
**Pull request by blackandred**

### CODE REVIEW

File: infracheck/checks/domain-expiration [hunk 1]
```diff
@@ -47,43 +56,77 @@ class DomainCheck:
...
 
-        while True:
+        while retries <= max_retries:
             try:
-                query = self.whois.query(self._domain)
+                _stdout_copy = sys.stdout
+
+                # the "whois" library is using print(), sys.stdout needs to be mocked for a moment to not print those
+                # messages
+                try:
```
Comment (krilmeer) [inline infracheck/checks/domain-expiration:70]: Is it possible to implement "with" clause instead? Not saying it's wrong but using try-finally to handle resources is a bit ugly, I'm just used to Java's (_try-with-resources_) or C#'s (_using_) way :)

Comment (blackandred) [reply infracheck/checks/domain-expiration:70]: It would be nice, but I cannot find such possibility there, but I always can be wrong.

Comment (blackandred) [reply infracheck/checks/domain-expiration]: Changed it a little bit.

File: infracheck/checks/domain-expiration [hunk 2]
```diff
@@ -29,16 +31,23 @@ import re
...
 from collections import namedtuple
+from whois._3_adjust import str_to_date
 
 
-ManualCheck = namedtuple('ManualCheck', 'expiration_date')
+WhoisResult = namedtuple('ManualCheck', 'expiration_date')
 
 
-class DomainCheck:
+class DomainCheck(object):
```
Comment (krilmeer) [inline infracheck/checks/domain-expiration:40]: What's the difference between no parameter and object parameter?

File: infracheck/checks/domain-expiration [hunk 3]
```diff
@@ -47,43 +56,79 @@ class DomainCheck:
...
             except Exception as e:
                 retries += 1
 
                 if retries > max_retries:
                     raise e
 
+                if isinstance(e, whois.exceptions.UnknownTld):
+                    shell_check_result = self._parse_shell_whois(self._domain)
+
+                    if not shell_check_result or not shell_check_result.expiration_date:
```
Comment (krilmeer) [inline infracheck/checks/domain-expiration:82]: Why `continue` if check doesn't return None?

Comment (blackandred) [reply infracheck/checks/domain-expiration:82]: To `retry` in the loop

File: infracheck/checks/domain-expiration (hunk 2 above)
Comment (blackandred) [reply infracheck/checks/domain-expiration:40]: Today - no reason, just looks better :joy: 
Historicaly: https://stackoverflow.com/a/45062077/6782994

File: infracheck/checks/domain-expiration [hunk 4]
```diff
@@ -47,43 +56,78 @@ class DomainCheck:
...
+        with open('/dev/null', 'w') as null_out:
+            try:
+                sys.stdout = null_out
+
+                return self.whois.query(domain)
+            finally:
+                sys.stdout = _stdout_copy
+
+    @staticmethod
+    def _ask_whois_in_shell(domain: str) -> str:
```
Comment (krilmeer) [inline infracheck/checks/domain-expiration:104]: What's happening here? How does output make it to the shell? Is it printed out by this library/class?

Comment (krilmeer) [reply infracheck/checks/domain-expiration:104]: Or is it a call to `whois` in terminal?

Comment (blackandred) [reply infracheck/checks/domain-expiration:104]: It is a spawned shell call, the output is captured from shell.

**Review by krilmeer — APPROVED** [review]
:+1:

**Pull request merged**